### DISCUSSION TITLE
How to reuse rule in different grammar file?

### DISCUSSION THREAD

**Discussion by appsubscribe** [pest-parser/pest#706] [Q&A] [answered]
I'd like to know in the case that we have many grammar files sharing the same custom rule. How can I re-use it?

a.pest
```txt
CUSTOM_ID = { 'a'..'z' }
```

b.pest
```txt
CUSTOM_ID = { 'a'..'z' }
```

Reading through the doc but still not sure how to do it. Thanks!

**Comment by tomtau** [comment, marked as answer]
It's not yet possible, but there's a WIP PR by @bobbbay : https://github.com/pest-parser/pest/pull/660

**Reply by appsubscribe** [reply]
Wow, thanks for a quick replied. Got it!

**Reply by tomtau** [reply]
One more WIP: https://github.com/pest-parser/pest/pull/758

**Reply by tomtau** [reply]
@appsubscribe v2.5.3 was just published: https://github.com/pest-parser/pest/discussions/767
It has included that patch which allows multiple grammars in derive statements. See the PR for more details: https://github.com/pest-parser/pest/pull/758

**Reply by pms1969** [reply]
I was just reaching for this ability to split out URI parsing into a separate file since the URI spec is....... rather extensive, and I just wanted to keep my main grammar clean..... https://www.rfc-editor.org/rfc/rfc3986#appendix-A

Thank you very much for your hard work. ❤️
